delta-proto/src/main/java/io/cdap/delta/proto/ColumnTransformation.java

```diff
@@ -43,10 +43,12 @@ public String getDirective() {
    */
   public void validate() {
     if (columnName == null || columnName.isEmpty()) {
-      throw new IllegalArgumentException("Column name of a column transformation should not be null or empty.");
+      throw new IllegalArgumentException(String.format("Column name of a ColumnTransformation for directive %s is " +
+                                                         "null or empty.", directive));
```

Comment:
seems like `directive` name can also be null or empty in which case this message will be confusing. Since we have this hierarchy table -> column -> directive in the configuration, it would be good to log table name for column validation error and table name and column name for directive validation errors.

Reply:
We can catch the IllegalArgumentException in the caller of validate and add table name as an additional context to exception message. Something like in `TableTransformation` class

```java
  public void validate() {
    if (tableName == null || tableName.isEmpty()) {
      throw new IllegalArgumentException(String.format("Table name of a TableTransformation that contains below " +
                                                         "ColumnTransformations : %s is null or empty.",
                                                       columnTransformations));
    }
    if (columnTransformations != null) {
      for (ColumnTransformation columnTransformation : columnTransformations) {
        try {
          columnTransformation.validate();
        } catch (IllegalArgumentException e) {
          throw new IllegalArgumentException(String.format("Column/Directive validation failed for table %s." +
                                                             " Reason: %s", tableName, e.getMessage()));
        }
      }
    }
} 
```

Reply:
that's a good idea:)

Reply:
done :)